### PR TITLE
fixed #667 retain scroll position when device orientation change

### LINKED ISSUE #667: remember scroll position on orientation change in translation
if you scroll in portrait mode to an ayah and switch to landscape, it returns to the top. we should at least return to the top of the ayah that was last being read.

### PR DESCRIPTION
**Pull request by hakimrie**
retain scroll position on translation page when device orientation change.

### CODE REVIEW

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/TranslationFragment.java [hunk 1]
```diff
@@ -39,6 +43,15 @@
...
   private QuranSettings mQuranSettings;
   private boolean mJustCreated;
+  private float mPercentLastScrollPosition;
+  private ViewTreeObserver.OnGlobalLayoutListener layoutListener =
+      new ViewTreeObserver.OnGlobalLayoutListener() {
+        @Override
+        public void onGlobalLayout() {
+          mTranslationView.scrollTo(0,
+              Math.round(mPercentLastScrollPosition * getTranslationViewHeight()));
+        }
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/TranslationFragment.java]: i'd actually remove it here also

**Comment by ahmedre** [comment]
my question to you is when this would matter - i guess only for translations, or also for when you accidentally move to portrait and back to landscape? if so, can we conditionally do this for translations (since in image mode in portrait, the entire screen fits the image, and only in landscape is it scrollable).

depending on this, if for translations, if we can figure out the first ayah on the screen easily (i have to look at the code, can't remember on top of my head), that might be better than a percentage since the percentage's meaning maybe different in portrait/landscape.

sorry for taking so long to reply to this

**Comment by hakimrie** [comment]
@ahmedre yes, this implementation is just for translations ( as defined in issue #667 ).
because current `TranslationView` is a `ScrollView`, I can't think any other way to save current position beside by using scroll listener (it will be different if we use `ListView/RecyclerView`).

using percentage make it possible to adjust scroll position to the right position. since the width/height of the screen different between landscape and portrait, we can't use absolute scroll position in portrait and applied it in landscape mode.

**Comment by ahmedre** [comment]
makes sense, jazakAllah khairan - i misread the patch the first time - looks good to me, just requesting that you remove the listener sooner as per the comment and then can merge.

**Comment by ahmedre** [comment]
(as an aside, that's another good point - in the future, this should use a `RecyclerView` instead of manually adding stuff to a scrolling window)

**Comment by hakimrie** [comment]
somehow, remove globallayoutlistener immediately after calling `scrollTo` didn't work, I think it because ayat/translation is added asynchronously using `TranslationTask`. I am thinking to refactor translationview to use `RecyclerView` instead of `ScrollView`. in the meantime I will close this PR.

**Comment by ahmedre** [comment]
awesome, looking forward to it, jazakAllah khairan!

**Pull request closed without merging**